scripts/repoint-frontend/repoint_frontend.sh

```diff
@@ -0,0 +1,269 @@
+  domain_status=$(aws apigateway get-domain-name --domain-name $domain_name)
+
+  if echo "$domain_status" | jq -re 'has("mutualTlsAuthentication")' > /dev/null 2>&1; then
+    aws apigateway update-domain-name \
+      --domain-name  $domain_name \
+      --patch-operations op=remove,path=/mutualTlsAuthentication/truststoreUri
+  fi
+fi
+
+if check_run_step "create new branch for proxy modifications"; then
+  current_branch=$(git rev-parse --abbrev-ref HEAD)
+  exit_on_failure "failed to get current git branch"
+
+  repsonse=""
+  info "current branch is: $current_branch"
+  query "do you want to use this branch? (y/n)"
+
+  if [[ "$response" != "y" ]]; then
+    error "checkout the branch you want to work with first"
+  fi
+
+  # create and checkout the new branch using the ticket ID
+  new_branch="CCM-${ticket_id}_REPOINT_DO_NOT_MERGE"
+  git checkout -b "$new_branch"
+  exit_on_failure "failed to checkout branch $new_branch"
+fi
+
+if check_run_step "modify proxy files"; then
+  output=$(python3 scripts/repoint-frontend/repoint_frontend.py $shortcode 2>&1)
+  exit_on_failure "python repoint script failed" "\n $output"
+  info_multiline "$output"
+fi
+
+commit_made=$false
+if check_run_step "stage, commit, and push changes"; then
+  git add proxies
+  exit_on_failure "filed to stage changes (git add)"
+
+  git commit -m "CCM-${ticket_id}: repointed frontend" --no-verify  # verify breaks on my machine, will fix this in due course
+  exit_on_failure "failed to commit changes (git commit)"
+
+  git push --set-upstream origin "$new_branch"
+  exit_on_failure "failed to push changes (git push)"
+  commit_made=$true
+fi
+
+# wait for mTLS update to complete
+if check_run_step "mTLS update - await domainName available status (or timeout)"; then
+  start_time=$(date +%s)
+  domain_name_available=$false
+  timeout_seconds=300
+  domain_name="comms-apim.de-$shortcode.communications.national.nhs.uk"
+
+  # timeout set to 300 seconds /5 minutes
+  while [ $(( $(date +%s) - start_time )) -lt $timeout_seconds ]; do
+    info_overwrite "waiting for available status (timeout in $(($timeout_seconds - ($(date +%s) - $start_time))) seconds)"
+
+    domain_status=$(aws apigateway get-domain-name --domain-name $domain_name)
+    exit_on_failure "aws sso session may have expired"
+
+    domain_name_status=$( echo "$domain_status" | jq -er '.domainNameStatus')
+
+    if [[ $domain_name_status == "AVAILABLE" ]]; then
+      domain_name_available=$true
+      echo ""  # prevents the previous info being overwritten
+      info_overwrite "mTLS update has completed"
+      break
+    fi
+
+    sleep 1
+  done
+  echo ""  # prevents the previous info being overwritten
+
+  if [[ $domain_name_available -eq $false ]]; then
+    warn "timeout reached before domain name became available - mTLS update is likely still processing"
+  fi
+fi
+
+# final output
+if [[ $commit_made -eq $true ]]; then
+  echo ""
```

Comment:
My fault for not updating these, there's a util function up top called spacer that does just this, you **could** either remove that or update the instances of `echo ""` to use it... but only if you want to because it doesn't really matter